http-client/src/main/java/io/micronaut/http/client/DefaultHttpClient.java

```diff
@@ -1316,32 +1315,31 @@ protected SslContext buildSslContext(URI uriObject) {
     /**
      * Configures the HTTP proxy for the pipeline.
      *
-     * @param pipeline     The pipeline
-     * @param proxyType    The proxy type
-     * @param proxyAddress The proxy address
+     * @param pipeline The pipeline
+     * @param proxy    The proxy
      */
-    protected void configureProxy(ChannelPipeline pipeline, Type proxyType, SocketAddress proxyAddress) {
+    protected void configureProxy(ChannelPipeline pipeline, Proxy proxy) {
```

Comment:
This would be a breaking change, could you restore the previous method and add a method that is overloaded?